addon/components/tri-state.js

```diff
@@ -30,60 +32,77 @@ export default Component.extend({
   layout,
 
   /**
-   * A service that is event aware so we can trigger actions in tri-state from external sources.
-   * Note this is NOT RECOMMENDED as it goes against DDAU and has the potential to introduce memory
-   * leaks if we don't clean up after ourselves. However, it can be useful in certain situations.
-   * @type {Object}
+   * Cache for the last resolved data to be shown during requests when showLastSuccessful is true
+   * @type {*}
+   * @private
    */
-  triEvents: inject.service(),
+  _lastResolvedData: null,
```

Comment:
I now see the issue you were talking about with re-implementing lastSuccessfulData with failFast promises that "fail" and display error components, but actually resolve. We can do it, the question just becomes is that feature useful? I think it could be in some scenarios so I'm a little reluctant to remove it.

Let's change this so we only cache successful responses that aren't settleAll "errors." It's a bit more work, but we'll continue to cover that use case.